### DISCUSSION TITLE
Would like info/advice for a site with baseUrl: "/docs/" and trailingSlash: false

### DISCUSSION THREAD

**Discussion by jharrell** [facebook/docusaurus#10086] [Q&A]
Hello there!

We recently moved [our docs site](https://github.com/prisma/docs) from Gatsby to Docusaurus. I'd first like to say that we're very happy with the maintainability with our new site. 

I would like to ask for reasoning behind the statement that [baseUrl always has a trailing slash](https://docusaurus.io/docs/api/docusaurus-config#baseUrl)

For our application, we would like no trailing slashes. This was the setup in our app previous to the switch to Docusaurus. It worked in most cases, except in cases where a link pointed to our baseUrl. In all cases, the link will be `/docs/` with a trailing slash. The only way we found is to use a raw `a` tag pointing to `/docs`. (I think that is due [to this code](https://github.com/facebook/docusaurus/blob/da2c0b4eae5d4a53f4cfe83c288a74fad9a83065/packages/docusaurus-utils-common/src/applyTrailingSlash.ts#L44-L48) but I'm not certain)

I understand why a baseUrl of `/` needs a trailing slash (pretty self evident) but I don't understand why a baseUrl of anything else requires one. Or, why a link to the baseUrl without a trailing slash is not allowed. 

Any advice or info here?

**Comment by slorber** [comment]
Thanks!

When you deploy a static site on a host under a baseurl, you generally deploy a subfolder containing that baseurl (for example `./myBaseUrl`). 

And when you deploy `./myBaseUrl/index.html`, hosts will generally serve it from `/myBaseUrl/` by default. Docusaurus has to emit files somewhere. We emit files in `./build` by default (`./build/index.html` is usually your home). If we wanted most hosts to not have a trailing slash for that "home URL", we'd have to emit a `../build/myBaseUrl.html` file, which is outside the `./build` folder and would be quite odd and wouldn't work with many hosts that generally require you to define a build subfolder to deploy.

Now, it's possible to configure hosts to serve `/myBaseUrl/` from `/myBaseUrl`, but this is host-specific and your responsibility to configure.

> Or, why a link to the baseUrl without a trailing slash is not allowed.

That's a different subject. The real question is: what is the canonical URL of your page (what Google sees and use as the "true" URL). You can link to that page with/without a trailing slash, but for consistency and SEO you'd rather always link to it through its canonical URL, which contains a trailing slash for the reasons above.

Our Link component has `<Link autoAddBaseUrl={false}>` option, although it's undocumented and I wouldn't particularly recommend it.

If you really want `/myBaseUrl` as a canonical URL then you could also use `baseUrl: '/'` and put your homepage at `./src/pages/myBaseUrl.js`. This way we'd output `./build/myBaseUrl.html` which remains "inside" your site and allows you to deploy the build folder.
